docs/source/auto_examples/example_uq_mcmc.py

```diff
@@ -0,0 +1,229 @@
+    weight_params={
+        "energy_weight_params": [0.0, 0.1],
+        "forces_weight_params": [0.0, 0.1],
+    }
+)
+# Read the dataset and compute the weight
+tset = Dataset(dataset_path, weight=weight)
+configs = tset.get_configs()
+
+# Create calculator
+calc = Calculator(model)
+ca = calc.create(configs)
+
+# Instantiate the loss function
+residual_data = {"normalize_by_natoms": False}
+loss = Loss(calc, residual_data=residual_data)
+
+# Train the model
+loss.minimize(method="L-BFGS-B", options={"disp": True})
+model.echo_opt_params()
+
+
+#########################################################################################
+# To perform MCMC simulation, we use :class:`~kliff.uq.MCMC`.This class interfaces with
+# ptemcee_ Python package to run PTMCMC, which utilizes the affine invariance property
+# of MCMC sampling. We simulate MCMC sampling at several different temperatures to
+# explore the effect of the scale of bias and overall error bars.
+
+
+# It is a good practice to specify the random seed to generate a reproducible simulation.
+seed = 1717
+np.random.seed(seed)
+
+# Define some variables that correspond to the dimensionality of the problem
+ntemps = 4  # Number of temperatures to simulate
+ndim = calc.get_num_opt_params()  # Number of parameters
+nwalkers = 2 * ndim  # Number of parallel walkers to simulate
+
+
+##########################################################################################
+# We start by instantiating :class:`~kliff.uq.MCMC`. This requires :class:`~kliff.loss.Loss`
+# instance to construct the likelihood function. Additionally, we can specify the prior
+# (or log-prior to be more precise), with the default be a uniform prior that is bounded
+# over a finite range that we specify, or otherwise it will retrieve the bounds from the
+# model (see :meth:`~kliff.models.KIMModel.set_opt_params`).
+#
+# To specify the temperatures to use, we can can use the arguments ``ntemps`` and
+# ``Tmax_ratio`` to specify how many temperatures to simulate and the ratio of the highest
+# temperature to the natural temperature :math:`T_0`, respectively. Alternatively, we can
+# also give a list of the temperatures to use via ``Tladder`` argument.
+#
+# .. note::
+#    It has been shown that including temperatures higher than :math:`T_0` helps the
+#    convergence of walkers sampled at :math:`T_0`
+
+
+# Create sampler
+sampler = MCMC(loss, ntemps=4, threads=4, random=np.random.RandomState(seed))
+
+# Initial starting point. This should be provided by the user.
+bounds = calc.get_opt_params_bounds()
+p0 = np.empty((4, 4, 2))
+for ii, bound in enumerate(bounds):
+    p0[:, :, ii] = np.random.uniform(*bound, (4, 4))
+
+# Run MCMC
+sampler.run_mcmc(p0, 10000)
+
+# Retrieve the chain
+chain = sampler.chain
+
+
+#########################################################################################
+# .. note::
+#    The initial states need to be an array with shape ``(K, L, N,)``, where ``K`` is the
```

Comment:
"The initial states $p_0$" to be more specific.

Reply:
I like this suggestion.